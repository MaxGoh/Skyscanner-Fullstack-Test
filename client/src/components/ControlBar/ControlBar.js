import React from 'react';
import BpkLargePriceAlertIcon from 'bpk-component-icon/lg/price-alerts';
import {
  BpkGridContainer,
  BpkGridRow,
  BpkGridColumn
} from 'bpk-component-grid';

import './ControlBar.scss';

const ControlBar = () => (
  <BpkGridContainer className='control-bar'>
    <BpkGridRow>
      <BpkGridColumn width={2}>
        Filter
      </BpkGridColumn>
      <BpkGridColumn width={2}>
        Sort
      </BpkGridColumn>
      <BpkGridColumn offset={3} width={5}>
        <span className='control-bar__price-alert'>
          <BpkLargePriceAlertIcon className='icon__price-alert' />
          Price alerts
        </span>
      </BpkGridColumn>
    </BpkGridRow>u
  </BpkGridContainer>
);

export default ControlBar;
