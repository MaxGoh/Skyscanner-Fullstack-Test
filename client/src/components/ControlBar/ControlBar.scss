@import "~bpk-mixins";

.control-bar {
  height: 48px;
  background: white;
  color: #00B2D6;

  &__price-alert {
    display: inline-block;
    float: right;
    svg {
      display: inline-block;
      vertical-align: middle;
    }
  }
}

.icon__price-alert {
  fill: $bpk-color-blue-500;
}