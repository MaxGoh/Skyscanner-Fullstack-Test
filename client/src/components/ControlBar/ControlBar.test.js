import React from 'react';
import ControlBar from './ControlBar';
import { shallow } from 'enzyme';

describe('<ControlBar />', () => {

  const component = shallow(<ControlBar />);

  it('should render properly without crashing', () => {
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('should have the text Filter', () => {
    expect(
      component
      .find('BpkGridColumn')
      .at(0)
      .contains('Filter')
    ).toBeTruthy();
  });

  it('should have the text Sort', () => {
    expect(
      component
      .find('BpkGridColumn')
      .at(1)
      .contains('Sort')
    ).toBeTruthy();
  });

  it('should have the text Price Alerts', () => {
    expect(
      component
      .find('.control-bar__price-alert')
      .contains(
       'Price alerts'
      )
    ).toBeTruthy();
  });

});
