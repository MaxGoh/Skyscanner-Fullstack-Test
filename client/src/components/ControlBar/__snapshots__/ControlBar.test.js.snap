// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlBar /> should render properly without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ControlBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <BpkGridRow
          className={null}
          padded={true}
        >
          <BpkGridColumn
            className={null}
            debug={false}
            mobileOffset={null}
            mobileWidth={null}
            offset={null}
            padded={true}
            tabletOffset={null}
            tabletWidth={null}
            width={2}
          >
            Filter
          </BpkGridColumn>
          <BpkGridColumn
            className={null}
            debug={false}
            mobileOffset={null}
            mobileWidth={null}
            offset={null}
            padded={true}
            tabletOffset={null}
            tabletWidth={null}
            width={2}
          >
            Sort
          </BpkGridColumn>
          <BpkGridColumn
            className={null}
            debug={false}
            mobileOffset={null}
            mobileWidth={null}
            offset={3}
            padded={true}
            tabletOffset={null}
            tabletWidth={null}
            width={5}
          >
            <span
              className="control-bar__price-alert"
            >
              <Unknown
                className="icon__price-alert"
              />
              Price alerts
            </span>
          </BpkGridColumn>
        </BpkGridRow>,
        "u",
      ],
      "className": "control-bar",
      "debug": false,
      "fullWidth": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <BpkGridColumn
              className={null}
              debug={false}
              mobileOffset={null}
              mobileWidth={null}
              offset={null}
              padded={true}
              tabletOffset={null}
              tabletWidth={null}
              width={2}
            >
              Filter
            </BpkGridColumn>,
            <BpkGridColumn
              className={null}
              debug={false}
              mobileOffset={null}
              mobileWidth={null}
              offset={null}
              padded={true}
              tabletOffset={null}
              tabletWidth={null}
              width={2}
            >
              Sort
            </BpkGridColumn>,
            <BpkGridColumn
              className={null}
              debug={false}
              mobileOffset={null}
              mobileWidth={null}
              offset={3}
              padded={true}
              tabletOffset={null}
              tabletWidth={null}
              width={5}
            >
              <span
                className="control-bar__price-alert"
              >
                <Unknown
                  className="icon__price-alert"
                />
                Price alerts
              </span>
            </BpkGridColumn>,
          ],
          "className": null,
          "padded": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Filter",
              "className": null,
              "debug": false,
              "mobileOffset": null,
              "mobileWidth": null,
              "offset": null,
              "padded": true,
              "tabletOffset": null,
              "tabletWidth": null,
              "width": 2,
            },
            "ref": null,
            "rendered": "Filter",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Sort",
              "className": null,
              "debug": false,
              "mobileOffset": null,
              "mobileWidth": null,
              "offset": null,
              "padded": true,
              "tabletOffset": null,
              "tabletWidth": null,
              "width": 2,
            },
            "ref": null,
            "rendered": "Sort",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="control-bar__price-alert"
              >
                <Unknown
                  className="icon__price-alert"
                />
                Price alerts
              </span>,
              "className": null,
              "debug": false,
              "mobileOffset": null,
              "mobileWidth": null,
              "offset": 3,
              "padded": true,
              "tabletOffset": null,
              "tabletWidth": null,
              "width": 5,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Unknown
                    className="icon__price-alert"
                  />,
                  "Price alerts",
                ],
                "className": "control-bar__price-alert",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "icon__price-alert",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Price alerts",
              ],
              "type": "span",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "u",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <BpkGridRow
            className={null}
            padded={true}
          >
            <BpkGridColumn
              className={null}
              debug={false}
              mobileOffset={null}
              mobileWidth={null}
              offset={null}
              padded={true}
              tabletOffset={null}
              tabletWidth={null}
              width={2}
            >
              Filter
            </BpkGridColumn>
            <BpkGridColumn
              className={null}
              debug={false}
              mobileOffset={null}
              mobileWidth={null}
              offset={null}
              padded={true}
              tabletOffset={null}
              tabletWidth={null}
              width={2}
            >
              Sort
            </BpkGridColumn>
            <BpkGridColumn
              className={null}
              debug={false}
              mobileOffset={null}
              mobileWidth={null}
              offset={3}
              padded={true}
              tabletOffset={null}
              tabletWidth={null}
              width={5}
            >
              <span
                className="control-bar__price-alert"
              >
                <Unknown
                  className="icon__price-alert"
                />
                Price alerts
              </span>
            </BpkGridColumn>
          </BpkGridRow>,
          "u",
        ],
        "className": "control-bar",
        "debug": false,
        "fullWidth": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <BpkGridColumn
                className={null}
                debug={false}
                mobileOffset={null}
                mobileWidth={null}
                offset={null}
                padded={true}
                tabletOffset={null}
                tabletWidth={null}
                width={2}
              >
                Filter
              </BpkGridColumn>,
              <BpkGridColumn
                className={null}
                debug={false}
                mobileOffset={null}
                mobileWidth={null}
                offset={null}
                padded={true}
                tabletOffset={null}
                tabletWidth={null}
                width={2}
              >
                Sort
              </BpkGridColumn>,
              <BpkGridColumn
                className={null}
                debug={false}
                mobileOffset={null}
                mobileWidth={null}
                offset={3}
                padded={true}
                tabletOffset={null}
                tabletWidth={null}
                width={5}
              >
                <span
                  className="control-bar__price-alert"
                >
                  <Unknown
                    className="icon__price-alert"
                  />
                  Price alerts
                </span>
              </BpkGridColumn>,
            ],
            "className": null,
            "padded": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Filter",
                "className": null,
                "debug": false,
                "mobileOffset": null,
                "mobileWidth": null,
                "offset": null,
                "padded": true,
                "tabletOffset": null,
                "tabletWidth": null,
                "width": 2,
              },
              "ref": null,
              "rendered": "Filter",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Sort",
                "className": null,
                "debug": false,
                "mobileOffset": null,
                "mobileWidth": null,
                "offset": null,
                "padded": true,
                "tabletOffset": null,
                "tabletWidth": null,
                "width": 2,
              },
              "ref": null,
              "rendered": "Sort",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <span
                  className="control-bar__price-alert"
                >
                  <Unknown
                    className="icon__price-alert"
                  />
                  Price alerts
                </span>,
                "className": null,
                "debug": false,
                "mobileOffset": null,
                "mobileWidth": null,
                "offset": 3,
                "padded": true,
                "tabletOffset": null,
                "tabletWidth": null,
                "width": 5,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Unknown
                      className="icon__price-alert"
                    />,
                    "Price alerts",
                  ],
                  "className": "control-bar__price-alert",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "className": "icon__price-alert",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "Price alerts",
                ],
                "type": "span",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "u",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
